.github/workflows/wasmcloud.yml

```diff
@@ -102,25 +102,33 @@ jobs:
             docker run --rm wasmcloud:$(nix eval --raw .#wasmcloud-x86_64-unknown-linux-musl-oci.imageTag) wasmcloud --version
 
     name: wasmcloud-${{ matrix.config.target }}
+    needs: check_for_changes
     runs-on: ubuntu-22.04
     steps:
     - uses: actions/checkout@v4.1.1
     - uses: ./.github/actions/install-nix
+      # need to run condition inside job steps so that job will pass if all steps are skipped
+      # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpull_requestpull_request_targetbranchesbranches-ignore
+      if: ${{ needs.check_for_changes.outputs.rust == 'true' }}
```

Comment:
This should run if anything in Nix flake changes, Rust source changes and if GitHub action spec itself has changed.
Relevant for other occurrences as well.

Reply:
ok, yeah good point, will update

Reply:
Ok, so I was already accounting for the flake changes in `.nix` but I've added github to that too. Combined them both into a `ci` check and added that into the rust check:
https://github.com/wasmCloud/wasmCloud/blob/5e57c5f7ca6d31c62f04fa9bda5683f0b45a801c/.github/workflows/_check-for-changes.yml#L74-L88

https://github.com/wasmCloud/wasmCloud/blob/5e57c5f7ca6d31c62f04fa9bda5683f0b45a801c/.github/workflows/_check-for-changes.yml#L44

https://github.com/wasmCloud/wasmCloud/blob/5e57c5f7ca6d31c62f04fa9bda5683f0b45a801c/.github/workflows/_check-for-changes.yml#L33-L35